packages/app/templates/ui-extensions/projects/checkout_ui_beta/README.md.liquid

```diff
@@ -0,0 +1,89 @@
+# Checkout UI Extension
+
+Checkout UI extensions let app developers build custom functionality that merchants can install at defined points in the checkout flow. You can learn more about checkout UI extensions in Shopify’s [developer documentation](https://shopify.dev/api/checkout-extensions/checkout).
+
+
+## Prerequisites
+Make sure to complete [all the prerequisites](https://shopify.dev/apps/checkout/delivery-instructions/getting-started#requirements) from any tutorials to develop a Checkout UI extension.
+
+## Tutorials
+The best way to get started is to follow some of the available Extension Tutorials:
+
+For creating an extension:
+* [Enable extended delivery instructions](https://shopify.dev/apps/checkout/delivery-instructions)
+* [Adding product offer recommendations](https://shopify.dev/apps/checkout/product-offers)
+* [Creating a custom banner](https://shopify.dev/apps/checkout/custom-banners)
+
+To add specific features to an extension
+* [Adding field validation](https://shopify.dev/apps/checkout/validation)
+* [Localizing an extension](https://shopify.dev/apps/checkout/localize-ui-extensions)
+
+## Getting Started
+Initially, your extension will have the following files:
+
+```
+└── my-app
+  └── extensions
+    └── my-checkout-ui-extension
+        ├── src
+        │   └── AdminCheckoutEditor.{{ srcFileExtension }} The source code for the ADMIN::CHECKOUT::EDITOR extension point
```

Comment:
I admit I haven't given this much thought, happy to bike shed.  Fortunately we can use whatever we want here, but since this will form the basis of future UI extensions it would be good to establish some conventions.

Objectives:

* It should be very easy to see which file is responsible for which extension point.
* It shouldn't be too unconventional.
* It should scale well for the most common amount of extension pints (approximately)

I won't cover subtle differences like uppercase vs lowercase first letter since we can iterate on that once we have the more macro level direction.

### Option 1: Use Extension point name without `::`
E.g: 

* `src/AdminCheckoutEditorRenderSettings.ts`
* `src/CheckoutDynamicRender.ts`

Pros:

* Matches files to extension points quite explicitly
* Flat structure is simple enough for most extensions
* Has a (weak) parallel with React [component conventions](https://github.com/Shopify/hydrogen/tree/v1.x-2022-07/packages/hydrogen/src/components)

Cons:

* Does loosing `::` in the filename make the connection with the extension point less obvious?

### Option 2: Use exactly the extension point name
E.g: 

* `src/Admin::CheckoutEditor::RenderSettings.ts`
* `src/Checkout::Dynamic::Render.ts`

Pros:

* Very explicitly matches files to extension points
* Flat structure is simple enough for most extensions

Cons:

* Quite unconventional
* Would the `::` cause us problems? (I don't think so, but I'm nervous)
* If an extension has MANY extension points would the flat structure benefit from some nesting?

### Option 3: Use nested folders

* `src/Admin/CheckoutEditor/RenderSettings.ts`
* `src/Checkout/Dynamic/Render.ts`

Pros:

* If extensions scale to MANY extension points the nesting might be useful
* Matches the rails way of nesting (I think this was the original inspiration)

Cons:

* Would 3rd party developers make the connection between `::` and the folder structure? Maybe not?
* if most extensions are only a few extension points ins this overkill? I suspect so.

----------

Thoughts?

Reply:
I think I like option 1 the best, personally. The names can get kind of long, but it is explicit and we can automate it, which is nice for moving fast. What do you think?

Reply:
Yeah, option 1 is my favourite too.

One question:  Is it `src/AdminCheckoutEditorRenderSettings.ts` (which I prefer) or `src/adminCheckoutEditorRenderSettings.ts` (lowercase a).